SuperteamDAO/earn
# FEAT: Updated bounties page to only consist of bounties that belongs to user's location and global

## What does this PR do?
This PR updates the code to filter bounties so that only those which belong to the user's location or are marked as global are shown to the user. This ensures that users see relevant bounties based on their location.

## Where should the reviewer start?
The reviewer should start by looking at the changes made in the /api/listings/index.tsx, particularly the OR clause is used. Ensure that the filtering logic using OR with region: 'GLOBAL' and region: location is correctly implemented.

## How should this be manually tested?

1. Set up your local environment to run the Next.js application.
2. Ensure you have the Prisma client configured and connected to the database.
3. Run the application and navigate to the page where bounties are listed.
4. Test the filtering by checking if bounties shown belong to either 'GLOBAL' or the specific user location.
5. You can change the user location to different values and verify if the filtering works as expected.

## Any background context you want to provide?
The goal of this update is to enhance the user experience by ensuring that users only see bounties that are relevant to them based on their location. Previously, users might have seen bounties from regions that were not applicable to them. This change makes the application more user-centric and improves engagement.

## What are the relevant issues?
There may have been previous issues or feature requests related to filtering bounties based on user location. If any specific issue numbers or tickets are related to this change, they should be referenced here (e.g., Issue #630 ).

## Screenshots (if appropriate)
![image](https://github.com/SuperteamDAO/earn/assets/80564174/7e5ad6e1-1088-4d60-be2f-6452f289183c)
As illustrated in the screenshot below, there are three bounties listed: one global, one from India, and one from the UK. With the new filtering logic applied, the bounties page will display only the global and India bounties, excluding the UK bounty: - 

![image](https://github.com/SuperteamDAO/earn/assets/80564174/f3a0103c-5de0-418b-b2bf-4a661ca73d01)
